tests/scripts/generate_code.pl

```diff
@@ -1,373 +0,0 @@
-#!/usr/bin/env perl
```

Comment:
The way test suites framework is implemented this script and template code is closely related. Since this PR changes templates this script is no longer compatible. Hence can't keep it.

Reply:
@sbutcher-arm I disagree with keeping both scripts. They have different APIs: the format of `.data` files has changed. It would be a significant effort to add a compatibility mode to one of the scripts and I don't see the point of doing it.